mlrun/artifacts/manager.py

```diff
@@ -246,6 +246,8 @@ def log_artifact(
                 # otherwise, we do not want to override it.
                 # this is mainly relevant for imported artifacts that have an explicit db_key value already set
                 db_key = item.db_key or key
+        if db_key != key:
```

Comment:
When `producer.kind == run` it will change the value of `db_key`. 
Please add a test to cover this issue